ctapipe/reco/HillasReconstructor.py

```diff
@@ -136,47 +191,44 @@ class are set to np.nan
         # filter warnings for missing obs time. this is needed because MC data has no obs time
         warnings.filterwarnings(action="ignore", category=MissingFrameAttributeWarning)
 
+        # Here we perform some basic quality checks BEFORE applying reconstruction
+        # This should be substituted by a DL1 QualityQuery specific to this
+        # reconstructor
+
         # stereoscopy needs at least two telescopes
         if len(hillas_dict) < 2:
+            result = ReconstructedShowerContainer()
             raise TooFewTelescopesException(
                 "need at least two telescopes, have {}".format(len(hillas_dict))
             )
-
         # check for np.nan or 0 width's as these screw up weights
         if any([np.isnan(hillas_dict[tel]["width"].value) for tel in hillas_dict]):
+            result = ReconstructedShowerContainer()
```

Comment:
Why create a result here and then raise an exception?

Reply:
because it was an older edit when I tried to move this into the call method, then I changed my mind, but I left it there

now it has been removed